### DISCUSSION TITLE
how to stop mongoose from writing schema defaults when using find?

### DISCUSSION THREAD

**Discussion by samislam** [Automattic/mongoose#12385] [Q&A] [answered]
I have defined these three models in mongoose:
```js
const clientSchema = new mongoose.Schema({
  name: String,
  role: String,
  age: {
     type: Number,
     default: 10
  }
})
clientSchema.pre('save', function(){
  const doc = this
  doc.role = 'client'
})
const adminSchema = new mongoose.Schema({
  name: String,
  role: String,
})
adminSchema.pre('save', function(){
  const doc = this
  doc.role = 'admin'
})
const userSchema = new mongoose.Schema({
  name: String,
  role: String,
  age: {
     type: Number,
     default: 10
  },
})
const AdminModel = mongoose.model('AdminModel', AdminModel, 'users')
const ClientModel = mongoose.model('ClientModel', ClientModel, 'users')
const UserModel = mongoose.model('UserModel', UserModel, 'users')
```

and I have the following route in my express app:

```js
app.route('/users').get(async(req, res, next)=>{
   const docs = UserModel.find({ role: 'admin' })
})
```

It works as expected, except that it's returning the following to the client-side:
```js
{
  "name": "James Breakers",
  "role": "admin",
  "age": 10
}
```

and this is wrong, and it causes many problems. 

On write, it works great, it saves it in the database without the `age: 10`, but the problem is when retrieving (reading) from the database, Mongoose is setting the field `age: 10`. 

and All of that is only because I used the UserModel to retrieve that admin. But I don't want that default behavior of Mongoose at all. 

How can I disable it? is there a schema option for that?

The question is also posted on StackOverflow as well
https://stackoverflow.com/questions/73583477/how-to-stop-mongoose-from-writing-schema-defaults-when-using-find-edited

**Comment by vkarpov15** [comment, marked as answer]
`const docs = await UserModel.find({ role: 'admin' }).setOptions({ defaults: false })` in Mongoose >= 5.12.10, see https://github.com/Automattic/mongoose/issues/7287

**Reply by samislam** [reply]
Great, is there a way we can do this on the schema level? by passing a property to the schema options?

**Reply by vkarpov15** [reply]
You would have to create a middleware.

```javascript
schema.pre('find', function() {
  this.setOptions({ defaults: false });
});
```
